cli_flags.go

```diff
@@ -38,6 +38,45 @@ const (
 	maxArgMapScaleFactor = 8
 )
 
+type additionalSymbolEndpoints []reporter.SymbolEndpoint
+
+func (s *additionalSymbolEndpoints) String() string {
+	if s == nil {
+		return ""
+	}
+	b, err := json.Marshal(s)
+	if err != nil {
+		return ""
+	}
+	return string(b)
+}
+
+func (s *additionalSymbolEndpoints) Set(value string) error {
+	if value == "" {
+		return nil
+	}
+	err := json.Unmarshal([]byte(value), s)
+	if err != nil {
+		return errors.New("invalid JSON")
+	}
+	for _, e := range *s {
+		if e.Site == "" || e.APIKey == "" || e.AppKey == "" {
+			return errors.New("site, api key and app key should all be set and non-empty strings")
+		}
+		if !isAPIKeyValid(e.APIKey) {
+			return fmt.Errorf("API key for site %s is not valid", e.Site)
+		}
+		if !isAPPKeyValid(e.AppKey) {
+			return fmt.Errorf("app key for site %s is not valid", e.Site)
```

Comment:
Shortened the error string, since urfave/cli includes more information by default when an error is encountered on `Set()` (e.g. name of the flag, etc..)